src/io.c

```diff
@@ -1333,9 +1347,14 @@ _dispatch_fd_entry_guarded_open(dispatch_fd_entry_t fd_entry, const char *path,
 	} else if (oflag & _O_TRUNC) {
 		dwCreationDisposition = TRUNCATE_EXISTING;
 	}
-	return (dispatch_fd_t)CreateFile(path, dwDesiredAccess,
+	int wide_len = MultiByteToWideChar(CP_UTF8, 0, path, -1, NULL, 0);
+	WCHAR *wide_path = malloc(wide_len * sizeof(WCHAR));
+	MultiByteToWideChar(CP_UTF8, 0, path, -1, wide_path, wide_len);
+	dispatch_fd_t fd = (dispatch_fd_t)CreateFileW(wide_path, dwDesiredAccess,
```

Comment:
Might be nice to maintain the style for windows and use `wszPath` for the path.

Reply:
Exactly! Something in `wide_path` bothered me and I even wasn't aware what exactly. Now I see (and can't unsee)🙈